src/main/java/org/broadinstitute/consent/http/service/EmailService.java

```diff
@@ -233,6 +241,71 @@ private void sendNewDARRequestEmail(
+    sendDARExpirationNotices();
+  }
+
+  private void sendDARExpirationNotices() {
+    EmailType emailType = EmailType.DAR_EXPIRED;
+    String interval = "1 year";
+    String noEmailForUserToWarnFoundLogTemplate = "User %d (%s) not found for expiring DAR.  Reference id: %s";
+    // Per value in ticket DT-1573
+    Timestamp minimumSubmittedDateForExpirations = Timestamp.from(Instant.ofEpochSecond(
+        LocalDate.of(2024, 9, 30).toEpochSecond(LocalTime.of(0, 0, 0, 0), ZoneOffset.UTC)));
+    List<DataAccessRequest> expiredDars = dataAccessRequestDAO.findAgedDARsByEmailTypeOlderThanInterval(
+        emailType.getTypeInt(), interval, minimumSubmittedDateForExpirations);
+    expiredDars.forEach(expiredDar -> {
+      try {
+        String referenceId = expiredDar.getReferenceId();
+        User user = userDAO.findUserById(expiredDar.getUserId());
+        String darCode = expiredDar.getDarCode();
+        String userName = user.getDisplayName();
+        if (user.getEmail() == null) {
+          // Do not throw here.  Log information about the DAR since this will continue
+          // to appear broken until manual intervention is taken to resolve the missing user
+          // email address
+          logWarn(String.format(noEmailForUserToWarnFoundLogTemplate,
+              expiredDar.getUserId(), userName, referenceId));
+        } else {
+          sendDarExpiredMessage(user, darCode, user.getUserId(), referenceId);
+        }
+      } catch (Exception e) {
+        logException(e);
+      }
+    });
+  }
+
+  private void sendDARExpirationReminderNotices() {
+    EmailType emailType = EmailType.DAR_EXPIRATION_REMINDER;
+    String interval = "11 months";
+    String noEmailForUserToWarnFoundLogTemplate = "User %d (%s) not found for expiring warning.  DAR reference id: %s";
+    // Per value in ticket DT-1573
+    Timestamp minimumSubmittedDateForExpirations = Timestamp.from(Instant.ofEpochSecond(
+        LocalDate.of(2024, 9, 30).toEpochSecond(LocalTime.of(0, 0, 0, 0), ZoneOffset.UTC)));
+    List<DataAccessRequest> expiredDars = dataAccessRequestDAO.findAgedDARsByEmailTypeOlderThanInterval(
+        emailType.getTypeInt(), interval, minimumSubmittedDateForExpirations);
+    expiredDars.forEach(expiredDar -> {
+      try {
+        String referenceId = expiredDar.getReferenceId();
+        User user = userDAO.findUserById(expiredDar.getUserId());
+        String darCode = expiredDar.getDarCode();
+        String userName = user.getDisplayName();
+        if (user.getEmail() == null) {
+          // Do not throw here.  Log information about the DAR since this will continue
+          // to appear broken until manual intervention is taken to resolve the missing user
+          // email address
+          logWarn(String.format(noEmailForUserToWarnFoundLogTemplate,
+              expiredDar.getUserId(), userName, referenceId));
+        } else {
+          sendDarExpirationReminderMessage(user, darCode, user.getUserId(), referenceId);
```

Comment:
Except for `emailType`, `interval`, and the type of `MailMessage` created, these methods appear to be the same. If so, can they be refactored to avoid this duplication?

Reply:
I had originally refactored this to do what you're suggesting only to move back to this because of different methods being used based on email type.  I could probably switch based on the value, but I'm not sure it will improve readability or reduce the lines of code in any meaningful way.

Reply:
I think I like it with the refactor.  Let me know what you think.